Fix 2GB-limit of the is_fragment(...) function.
Applies to squashfs-tools 4.3.
Signed-off-by: Guan, Xin <guanx.bac@...>
diff -Naurdp squashfs4.3.orig/squashfs-tools/mksquashfs.c squashfs4.3/squashfs-tools/mksquashfs.c
--- squashfs4.3.orig/squashfs-tools/mksquashfs.c	2014-05-13 00:18:20.000000000 +0200
+++ squashfs4.3/squashfs-tools/mksquashfs.c	2014-09-15 09:55:02.228930272 +0200
@@ -2055,7 +2055,7 @@ struct file_info *duplicate(long long fi
 
 inline int is_fragment(struct inode_info *inode)
 {
-	int file_size = inode->buf.st_size;
+	off_t file_size = inode->buf.st_size;
 
 	/*
 	 * If this block is to be compressed differently to the
